### PR TITLE
Drop json-jwt

### LINKED ISSUE #68: Parse error reading JWS
I'm basically running through readme and I'm not sure why this is happening. (using any other email address doesn't help)

```
require 'openssl'
private_key = OpenSSL::PKey::RSA.new(4096)
endpoint = "https://acme-staging.api.letsencrypt.org/"

require 'acme-client'
client = Acme::Client.new(private_key: private_key, endpoint: endpoint)
registration = client.register(contact: 'mailto:contact@example.com')
Acme::Client::Error::Malformed: Parse error reading JWS
        from /usr/lib64/ruby/gems/2.2.0/gems/acme-client-0.3.1/lib/acme/client/faraday_middleware.rb:35:in `raise_on_error!'
        from /usr/lib64/ruby/gems/2.2.0/gems/acme-client-0.3.1/lib/acme/client/faraday_middleware.rb:25:in `on_complete'
        from /usr/lib64/ruby/gems/2.2.0/gems/acme-client-0.3.1/lib/acme/client/faraday_middleware.rb:12:in `block in call'
        from /usr/lib64/ruby/gems/2.2.0/gems/faraday-0.9.2/lib/faraday/response.rb:57:in `on_complete'
        from /usr/lib64/ruby/gems/2.2.0/gems/acme-client-0.3.1/lib/acme/client/faraday_middleware.rb:12:in `call'
        from /usr/lib64/ruby/gems/2.2.0/gems/faraday-0.9.2/lib/faraday/rack_builder.rb:139:in `build_response'
        from /usr/lib64/ruby/gems/2.2.0/gems/faraday-0.9.2/lib/faraday/connection.rb:377:in `run_request'
        from /usr/lib64/ruby/gems/2.2.0/gems/faraday-0.9.2/lib/faraday/connection.rb:177:in `post'
        from /usr/lib64/ruby/gems/2.2.0/gems/acme-client-0.3.1/lib/acme/client.rb:25:in `register'
        from (irb):9
        from /usr/bin/irb:11:in `<main>'
```

### LINKED ISSUE #75: replace json-jwt

### PR DESCRIPTION
**Pull request by jhass**
Fixes #75 closes #68 closes #33 

I didn't rerecord the cassettes but I manually verified that everything still works. VCR by default matches only against the request method and URI, matching against the body is rather difficult for us due to non-deterministic key usage and random hostnames in the specs, so there's no real benefit to rerecording the cassettes here.

The implementation is doing just what we need, not anything more. Still it should produce identical results to json-jwt for our usecases.

### CODE REVIEW

File: lib/acme/client.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
+# frozen_string_literal: true
```
Comment (unixcharles) [inline lib/acme/client.rb]: fancy 👌

**Comment by unixcharles** [comment]
awesome 👍 

@mac-adam-chaieb mind to give a second pair of 👀 ?

File: lib/acme/client/resources/challenges/dns01.rb [hunk 2]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
 class Acme::Client::Resources::Challenges::DNS01 < Acme::Client::Resources::Challenges::Base
   CHALLENGE_TYPE = 'dns-01'.freeze
```
Comment (mac-adam-chaieb) [inline lib/acme/client/resources/challenges/dns01.rb]: Do you still need the `freeze` on top of `frozen_string_literal: true`? Not sure how that works.

**Comment by mac-adam-chaieb** [comment]
I don't know how `frozen_string_literal` works, but there were a couple `freeze` calls that seemed redundant to me, but I could be wrong. Crypto stuff is out of my knowledge.

That being said, LGTM.

**Comment by jhass** [comment]
The comment only works for Ruby 2.3+, so those are still useful.

**Pull request merged**